app/code/Magento/Checkout/Model/Session.php

```diff
@@ -588,4 +583,22 @@ protected function isQuoteMasked()
     {
         return $this->isQuoteMasked;
     }
+
+    /**
+     * Returns quote for customer if there is any
+     */
+    private function getQuoteByCustomer(): ?CartInterface
```

Comment:
Do you think this function could be made a `protected` instead?

Reply:
Could you explain why it’s needed ?

Reply:
Because if there is a need to extend the class, then this method is accessible in the child. If not this method will not be available and it will require setting a preference. Plus, it would maintain consistency as all the other methods are protected in the class.

Reply:
Extending customer session is definitely not a good thing and in core we shouldn’t allow that. If you really need it - you could just copy the method to your class